src/scaffoldmaker/meshtypes/meshtype_3d_brainstem.py

```diff
@@ -0,0 +1,376 @@
+        cylinderCentralPath = CylinderCentralPath(region, centralPath, elementsCountAlong)
+
+        cylinderShape = CylinderShape.CYLINDER_SHAPE_FULL if full else CylinderShape.CYLINDER_SHAPE_LOWER_HALF
+
+        taperedParams = Tapered(majorRatio=options['Taper major increment'],
+                                minorRatio=options['Taper minor increment'],
+                                majorProgressionMode=ConeBaseProgression.ARITHMETIC_PROGRESSION,
+                                minorProgressionMode=ConeBaseProgression.ARITHMETIC_PROGRESSION)
+
+        base = CylinderEnds(elementsCountAcrossMajor, elementsCountAcrossMinor,
+                            centre=[0.0, 0.0, 0.0],
+                            alongAxis=cylinderCentralPath.alongAxis[0], majorAxis=cylinderCentralPath.majorAxis[0],
+                            minorRadius=cylinderCentralPath.minorRadii[0])
+
+        cylinder1 = CylinderMesh(fm, coordinates, elementsCountAlong, base,
+                            cylinderShape=cylinderShape,
+                                 tapered = taperedParams,
+                                 cylinderCentralPath=cylinderCentralPath, useCrossDerivatives=False)
+
+        iRegionBoundaries = [int(7*elementsCountAlong/15),int(14*elementsCountAlong/15)]
+        for elementIdentifier in range(1, mesh.getSize()+1):
+            element = mesh.findElementByIdentifier(elementIdentifier)
+            brainstemMeshGroup.addElement(element)
+            if elementIdentifier > (iRegionBoundaries[-1]*elemPerLayer):
+                midbrainMeshGroup.addElement(element)
+            elif elementIdentifier > (iRegionBoundaries[0]*elemPerLayer) and elementIdentifier <= (iRegionBoundaries[-1]*elemPerLayer):
+                ponsMeshGroup.addElement(element)
+            else:
+                medullaMeshGroup.addElement(element)
+
+        ##############################
+        # point markers
+        ##############################
+        eIndexPM = {}
+        xiPM = {}
+        pointMarkers = {}
+        eIndexPM['caudal-dorsal'] = int(elemPerLayer/2)
+        eIndexPM['midRostralCaudal-dorsal'] = int(elemPerLayer / 2) + (elemPerLayer * int((elementsCountAlong/2)-1))
+        eIndexPM['rostral-dorsal'] = (elemPerLayer*(elementsCountAlong-1)) + int(elemPerLayer/2)
+        eIndexPM['caudal-ventral'] = int(elemPerLayer/2) - (elementsCountAcrossMinor-1)
+        eIndexPM['midRostralCaudal-ventral'] = eIndexPM['midRostralCaudal-dorsal'] - (elementsCountAcrossMinor-1)
+        eIndexPM['rostral-ventral'] = int((elemPerLayer*(elementsCountAlong-1)) + (int(elemPerLayer/2) - elementsCountAcrossMinor + 1))
+        xiPM['caudal-ventral'] = [1.0, 0.0, 0.0]
+        xiPM['caudal-dorsal'] = [1.0, 0.0, 1.0]
+        xiPM['midRostralCaudal-ventral'] = [1.0, 1.0, 0.0]
+        xiPM['midRostralCaudal-dorsal'] = [1.0, 1.0, 1.0]
+        xiPM['rostral-ventral'] = [1.0, 1.0, 0.0]
+        xiPM['rostral-dorsal'] = [1.0, 1.0, 1.0]
+        for key in eIndexPM.keys():
```

Comment:
We're making the marker points as annotationGroups now. This makes the code cleaner as there isn't the risk of not matching by name, however it means we must request ILX terms from K-core; mainly requires putting the name and description on a Google spreadsheet.
These names are not sufficient however, as they must be qualified with 'Brainstem', 'midline' and 'point' to be fully prescriptive. Is 'rostral' or 'cranial' more appropriate here -- we need terms that make sense across species.
The names should be plain english, uncapitalised without camelCase.
I don't think we need triple qualified rostral+caudal+ventral/dorsal as only 2 should be sufficient (along with midline).
Wouldn't the midRostralCaudal* points be better on the medulla-pons boundary?
Alternatively to many of these points, we could define a midline group around the boundary...

Reply:
I am unsure what you want here. I think it's probably better to talk to the next mapper to make these changes.